internal/instrument/call.go

```diff
@@ -0,0 +1,50 @@
+package instrument
```

Comment:
Had to introduce this new package under `internal` to break what would have been a circular package dependency between `api` and `internal/metrics`. I think it's a nice result though.